### ISSUE TITLE
[naming-conventions] The camelcase naming doesn't fit React functional component's convention

### ISSUE THREAD

**Issue by alvis** [typescript-eslint/typescript-eslint#2246]
Perhaps quite specific to React, but it's very common and recommended for any stateless functional component to be named in pascal case even they're just arrow functions. For example:

```ts
// expected: OK
// current behaviour: Variable name `App` must match one of the following formats: camelCase, UPPER_CASE
const App: React.FC<{ message: string }> = ({ message }) => (
  <div>{message}</div>
);
```

Clearly, the problem is that in the current `naming-conventions` setting, any arrow function should be in camelCase, which is almost universally true except for React. There was a discussion in #722, but it lacks a follow-up.

In generally, it's okay. But for any react project, I've to unwillingly disable naming-conventions or otherwise I'll get tons of warning...

**Comment by bradzacher** [comment]
have you considered using the `types: ['function']` selector to apply a convention to variables that define a function?

https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/naming-convention.md#selector-options

**Comment by alvis** [comment]
@bradzacher With `types: ['function']`, now all functional const can be set to accept both camel case can pascal case.
It's not ideal, but it does do the job. Thx.
